pkg/apis/tensorflow/v1beta2/types.go

```diff
@@ -42,6 +42,16 @@ type TFJob struct {
 
 // TFJobSpec is a desired state description of the TFJob.
 type TFJobSpec struct {
+	// Specifies the duration in seconds relative to the startTime that the job may be active
+	// before the system tries to terminate it; value must be positive integer
+	// +optional
+	ActiveDeadlineSeconds *int64 `json:"activeDeadlineSeconds,omitempty"`
```

Comment:
I wonder if these should be part of the TFJobSpec, or refactored to the common types. We should be able to just create a generic "JobSpec" type that contains ActiveDeadlineSeconds, BackoffLimit, CleanPodPolicy, TTLSeconds, and some generic ReplicaSpecs map. Then each individual operator just needs to define and implement its replica specs and reconcile logic.

Not necessary for this PR of course, just some thoughts. @johnugeorge what do you think?

Reply:
I agree. We should take this up when we restructure code into common repo( post 0.5 release).  I see that there are lots of common code that could be shared across.